backend/schemas.py

```diff
@@ -28,29 +29,107 @@ class AttributeDefinitionBase(BaseModel):
     key: bool
     description: Optional[str]
     bind_to_schema: Optional[int]
-    
+
+    class Config:
+        orm_mode = True
+        allow_population_by_field_name = True
+
 
 class AttrDefSchema(AttributeDefinitionBase):
-    attr_id: int
+    attr_id: int = Field(alias='attribute_id')
 
 
 class AttrDefWithAttrDataSchema(AttributeDefinitionBase, AttributeCreateSchema):
-    pass
+    @classmethod
+    def from_orm(cls, obj: Any):
+        if isinstance(obj, AttributeDefinition):
+            obj.type = obj.attribute.type.name
+            obj.name = obj.attribute.name
+        return super().from_orm(obj)
 
 
 class AttributeDefinitionUpdateSchema(AttributeDefinitionBase):
-    id: int
+    attr_def_id: int
+
+
+class AttributeDefinitionUpdateWithNameSchema(AttributeDefinitionBase):
+    name: str
+
+
+def validate_slug(cls, slug: str):
+    if re.match('^[a-zA-Z0-9]+(?:-[a-zA-Z0-9]+)*$', slug) is None:
```

Comment:
I don't think you need the `?:`. A capturing group should work just as well.

After looking at the handling of `id_or_slug` in the CRUD functions, I'm afraid we need to address one issue with this pattern:

The pattern allows numeric slugs like e.g. `123`. If my understanding is correct, the CRUD functions would treat this as ID instead of a slug. We might need a slightly more complicated pattern, e.g.:

```suggestion
    if re.match('^([a-zA-Z][-a-zA-Z0-9]+)|([0-9]+[-a-zA-Z]+)$', slug) is None:
```

Reply:
> A capturing group should work just as well

Yeah, it seems to be so

> would treat this as ID instead of a slug

Right, haven't thought about it. I'll change regexp to `^[a-zA-Z]+(-[a-zA-Z0-9]+)*$`, you can take a look at it here https://regex101.com/r/92FRVJ/2

Reply:
Looking at your test strings, I would object strongly only to `456` and those starting with `-`; maybe even `hello-$-world`.

If you want to exclude slugs starting with a number, it's also fine.

Reply:
> object strongly only to...

Well, I don't want to allow slugs that have multiple `-` in a row or which end with `-` because they just look wrong to me

> If you want to exclude slugs starting with a number

Yes, that was my idea

Reply:
That will work for me :+1: